validation_results.py

```diff
@@ -0,0 +1,183 @@
+from sqlalchemy import create_engine, Integer, String, Column, DateTime, ForeignKey, JSON, Boolean
```

Comment:
Current database.py for reference: https://github.com/buildingSMART/validate/blob/release/application/database.py

I think a possible solution to this is the current database.py importing what is needed from validation_results.py 

So something like:

In database.py:

```
from validation_results import ValidationOutcome, OutcomeSeverity

class user(Base, Serializable):
   #definition
   
```
In validation_results.py:

```
class ValidationOutcome(Base, Serializable):
   #definition
```


The reason for this is that in the validate repository database.py we will need a lot of classes that are kind of unrelated to gherkin, like for example user, model, bsdd and perhaps many others. 

The drawback is that we will have to keep the classes in gherkin validation_results.py consistent with validate database.py, but this is something to do anyway. 

What do you think?

Reply:
Good nuanced answer, I tbh don't know. Let's discuss in the next meeting as it also impacts the rewrite to Django. In the end both models need to be mapped to the same database so that probably gives us some constraints that I also don't know.

Reply:
Hm thinking about this a bit longer. I do think everything should be in the webapp's database.py, because we spent effort in generalizing this for all the check types.

Reply:
I don't mind that solution. 

But how do we do it from the practical perspective, as there's no database.py in this repo? Do you have something in mind or should we brainstorm it a little?

Reply:
One thought from somewhat of the "outsider's" perspective: how about combining `ifc-gherkin-rules` into the `validate` repo to create a single monorepo?  Are there specific benefits of keeping the former as a submodule in the latter?  In addition to the database model consideration mentioned above it would also potentially help consolidate feedback and updates into a single location for issues, pull requests, etc.  This would also help in terms of documentation, where everything related to the Validation Service could potentially be in one spot.

Reply:
> Are there specific benefits of keeping the former as a submodule in the latter

I think the main benefit was the ability to run the gherkin module separately. I'd think this would also be possible when combining everything into a single monorepo. A disadvantage would perhaps be that redundant database tables (e.g. related to the bsdd or syntax task) are set up when running the gherkin-part with the same database. I'm not sure whether this would be really a problem, apart from it being a bit messy.
It would also mean we'd probably have to do the same for the syntax and bsdd part. 

>everything related to the Validation Service could potentially be in one spot
potentially help consolidate feedback and updates into a single location for issues, pull requests_

Agreeing with this part. We have Jira now for the task, and pull requests and issues scattered over multiple repositories. Technical documentation in one spot also sounds very good.

For now, I'd say for now it's the most practical, with conscientiousness, to manually create the same database.py file for both the validate and the gherkin repos. And discuss merging the repo's after the second week of January.

Reply:
Hi guys!

Thank you for the comments! I'm sorry, as I discussed this a little bit with Thomas on Slack, without updating this thread. 

As the outcome of this discussion, I've trimmed the amount of DB code in the gherkin repository and came up with more or less this boiler plate code:

```
# sys.path.append(r"PATH TO VALIDATE") # TODO -> add the path if necessary
try:
    from validate_repository import OutcomeSeverity, ValidationOutcome, ValidationOutcomeCode 
except (ModuleNotFoundError, ImportError):
    from gherkin_repository import OutcomeSeverity, ValidationOutcome, ValidationOutcomeCode
```

The idea behind this is to try importing the necessary code from validate repository, which will be the main source of DB related code. This will work well in production. 

In the gherkin repository, this try will fail, but in that case we'll use the local gherkin definitions. 

This is not the most elegant, but it is quite difficult to do in an elegant way, especially since there are a number of moving parts right now.

Reply:
We know the decision we take, which is using the git submodule. I'll close this conversation.